### PR TITLE
[Thesaurus] Updating synonyms via a new dedicated REST service.

### PR DESCRIPTION
**Pull request by NicolasEYSSERIC**
Improving user experience too...

### CODE REVIEW

File: core-war/src/main/java/org/silverpeas/web/thesaurus/servlets/ThesaurusRequestRouter.java [hunk 1]
```diff
@@ -36,9 +36,11 @@
 import javax.servlet.http.HttpServletRequest;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.HashMap;
 import java.util.Hashtable;
```
Comment (silveruser) [inline core-war/src/main/java/org/silverpeas/web/thesaurus/servlets/ThesaurusRequestRouter.java]: MINOR Remove this unused import 'java.util.Hashtable'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

File: core-war/src/main/java/org/silverpeas/web/thesaurus/servlets/ThesaurusRequestRouter.java [hunk 2]
```diff
@@ -36,9 +36,11 @@
...
 import java.util.Iterator;
```
Comment (silveruser) [inline core-war/src/main/java/org/silverpeas/web/thesaurus/servlets/ThesaurusRequestRouter.java]: MINOR Remove this unused import 'java.util.Iterator'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by silveruser** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ThesaurusSessionController.java#L351](https://github.com/Silverpeas/Silverpeas-Core/blob/cd3aae44fd793851f11a9145d4977e470621aeff/core-war/src/main/java/org/silverpeas/web/thesaurus/control/ThesaurusSessionController.java#L351): Remove this unused private "getNewSynonym" method. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by SilverYoCha** [comment]
Merged.
Closed manually because of manual git operations performed during integration.

**Pull request closed without merging**